### PR TITLE
Cleaning up docker build process

### PR DESCRIPTION
**Pull request by sundargates**
### Context

I’d like to outline the goals for this PR.

Firstly, it aims to integrate the docker build process with the build system instead of having it defined in various places such as shell, gradle, and the dockerfile.
Secondly, it aims to set up the server with a few job clusters, making it easier to launch these clusters with the helm setup.

### Checklist

- [ ] `./gradlew build` compiles code correctly
- [ ] Added new tests where applicable
- [ ] `./gradlew test` passes all tests
- [ ] Extended README or added javadocs where applicable

### CODE REVIEW

File: mantis-control-plane/mantis-control-plane-server/build.gradle [hunk 1]
```diff
@@ -62,6 +63,43 @@ dependencies {
     testImplementation testFixtures(project(":mantis-control-plane:mantis-control-plane-core"))
     testImplementation libraries.commonsIo
 }
+def installDir = file("${buildDir}/install")
+def controlPlaneState = file("${project.resources}/docker")
+def ci = System.getenv('GITHUB_ACTIONS')
+def imageRepository = ci ? 'netflixoss' : 'localhost:5001/netflixoss'
```
Comment (calvin681) [inline mantis-control-plane/mantis-control-plane-server/build.gradle:69]: Add a comment here that we assume local docker registry running on 5001

**Review by calvin681 — COMMENTED** [review]
I have to say, this is quite genius.

File: mantis-server/mantis-server-agent/build.gradle [hunk 2]
```diff
@@ -37,13 +38,43 @@ dependencies {
...
+        // copy the files from the build context to the image
+        instruction 'COPY mantis-server-agent/bin/* /apps/mantis/mantis-server-agent/bin/'
+        instruction 'COPY mantis-server-agent/lib/* /apps/mantis/mantis-server-agent/lib/'
+        instruction 'COPY distributions/* mantis-artifacts/storage/'
+        instruction 'COPY resources/* conf/'
+        entryPoint 'bin/mantis-server-agent', '-p', 'conf/agent-docker.properties'
+    }
+
+    javaApplication {
+        baseImage = 'azul/zulu-openjdk:8-latest'
```
Comment (calvin681) [inline mantis-server/mantis-server-agent/build.gradle:69]: This comment is not related to this PR. Can you try changing to this to jdk17 and see what happens when you launch with helm?

**Review by calvin681 — APPROVED** [review]

File: mantis-control-plane/mantis-control-plane-server/build.gradle [hunk 3]
```diff
@@ -19,6 +19,7 @@ plugins {
 }
 
 apply plugin: 'application'
+apply plugin: 'com.bmuschko.docker-java-application'
```
Comment (Andyz26) [inline mantis-control-plane/mantis-control-plane-server/build.gradle:22]: Shall we lock version on this plugin too?

**Review by Andyz26 — COMMENTED** [review]
looks like path resolution needs some refactor in testcontainers.

File: mantis-control-plane/mantis-control-plane-server/build.gradle [hunk 4]
```diff
@@ -62,6 +63,43 @@ dependencies {
...
+        instruction 'COPY mantis-control-plane-server/bin/* bin/'
+        instruction 'COPY mantis-control-plane-server/lib/* lib/'
+        instruction 'COPY resources/* conf/'
+        instruction 'COPY docker/job-clusters/* job-clusters/'
+        instruction 'COPY docker/scripts/* bin/'
+
+        entryPoint './bin/start.sh'
+    }
+
+    javaApplication {
```
Comment (calvin682) [inline mantis-control-plane/mantis-control-plane-server/build.gradle:94]: `javaApplication` will create an entryPoint in the generated Dockerfile. And then the previous `dockerCreateDockerfile` section will also create an entryPoint. is that OK? I'm not sure if the generated Dockerfile is actually valid.

Comment (sundargates) [reply mantis-control-plane/mantis-control-plane-server/build.gradle:94]: Yeah, the second entry point overrides the first one. I have tested it end to end, and I can give a demo if you would like.
